bundle-generator/deployment/src/main/java/io/quarkiverse/operatorsdk/bundle/deployment/builders/CsvManifestsBuilder.java

```diff
@@ -327,11 +329,16 @@ private void handlerPermission(List<PolicyRule> rules,
             String serviceAccountName,
             NamedInstallStrategyFluent.SpecNested<ClusterServiceVersionSpecFluent.InstallNested<ClusterServiceVersionFluent.SpecNested<ClusterServiceVersionBuilder>>> installSpec) {
         if (!rules.isEmpty()) {
-            installSpec
-                    .addNewPermission()
-                    .withServiceAccountName(serviceAccountName)
-                    .addAllToRules(rules)
-                    .endPermission();
+            if (installSpec.hasMatchingPermission(permissionHasServiceAccountName(serviceAccountName))) {
```

Comment:
Use a local variable for the result of `permissionHasServiceAccountName` to avoid creating it twice.

Reply:
Done